pipeline.py

```diff
@@ -28,61 +36,116 @@ class MorphologicalToken:
+        self.mystem_tags = ''
+        self.pymorphy_tags = ''
 
     def __str__(self):
-        return "MorphologicalToken instance here"
+        return self.normalized_form + f'<{self.mystem_tags}>' + f'({self.pymorphy_tags})'
 
 
 class CorpusManager:
     """
     Works with articles and stores them
     """
     def __init__(self, path_to_raw_txt_data: str):
-        pass
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = self._scan_dataset()
 
     def _scan_dataset(self):
         """
         Register each dataset entry
         """
-        pass
+        path = Path(self.path_to_raw_txt_data)
+        all_files = [str(f_path) for f_path in path.iterdir()]
+        raw_files = {}
+
+        for file in all_files:
+            if '_raw.txt' in file:
+                try:
+                    text_id = int(file.split('\\')[-1].split('_')[0])
+                except ValueError:
+                    text_id = int(file.split('/')[-1].split('_')[0])
+
+                raw_files[text_id] = Article(url=None, article_id=text_id)
+
+        return raw_files
 
     def get_articles(self):
         """
         Returns storage params
         """
-        pass
+        return self._storage
 
 
 class TextProcessingPipeline:
     """
     Process articles from corpus manager
     """
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
+        self._text = ''
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+
+        storage = self.corpus_manager.get_articles()
+
+        for article in storage.values():
+            self._text = article
+            processed_text = list(map(str, self._process()))
+            article.save_processed(' '.join(processed_text))
 
     def _process(self) -> List[type(MorphologicalToken)]:
         """
         Performs processing of each text
         """
-        pass
+        text = self._text.get_raw_text()
+        result = Mystem().analyze(text)
+        tokens = []
+
+        for word in result:
+
+            if 'analysis' in word and word['analysis']:
+                token = MorphologicalToken(original_word=word, normalized_form=word['analysis'][0]['lex'])
+                token.mystem_tags = word['analysis'][0]['gr']
+
+                morph = pymorphy2.MorphAnalyzer()
+                pymorphy_tags = morph.parse(word['text'])
+                token.pymorphy_tags = pymorphy_tags[0].tag
+
+                tokens.append(token)
+
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path = Path(path_to_validate)
+
+    if not path.exists():
+        raise FileNotFoundError
+
+    if not path.is_dir():
+        raise NotADirectoryError
+
+    if not list(path.iterdir()):
+        raise EmptyDirectoryError
+
+    if len(list(path.iterdir())) % 2:
```

Comment:
maybe working, but make the check explicit and add check for balanced including numbers validation